File: src/main/java/com/orange/dojo/elephantcarpaccio/terminal/NumberReader.java
package com.orange.dojo.elephantcarpaccio.terminal;

import java.util.InputMismatchException;
import java.util.Scanner;

public class NumberReader {
  private final Scanner input;

  public NumberReader(Scanner textInputScanner) {
    this.input = textInputScanner;
  }

  public int readInteger() {
    try {
      return input.nextInt();
    } catch (InputMismatchException e) {
      throw new InputMismatchException("Input value is not an integer");
    }
  }

}

File: kotlin/src/main/kotlin/com/orange/dojo/elephantcarpaccio/terminal/NumberReader.kt
package com.orange.dojo.elephantcarpaccio.terminal


import java.util.*

internal class NumberReader(private val input: Scanner) {

    fun readInteger(): Int {
        try {
            return input.nextInt()
        } catch (e: InputMismatchException) {
            throw InputMismatchException("Input value is not an integer")
        }

    }

    fun readFloat(): Float {
        try {
            return input.nextFloat()
        } catch (e: InputMismatchException) {
            throw InputMismatchException("Input value is not a float")
        }

    }
}
